runtimes/web/src/ui/app.ts

```diff
@@ -373,9 +374,17 @@ export class App extends LitElement {
                 return; // Browser doesn't support gamepads
             }
 
-            for (const gamepad of navigator.getGamepads()) {
-                if (gamepad == null || gamepad.mapping != "standard") {
-                    continue; // Disconnected or non-standard gamepad
+            for (const [i, gamepad] of navigator.getGamepads().entries()) {
+                if (gamepad == null) {
+                    continue; // Disconnected gamepad
+                } else if (gamepad.mapping != "standard") {
+                    // The gamepad is available, but nonstandard, so we don't actually know how to read it.
+                    // Let's warn once, and not use this gamepad afterwards.
+                    if (!this.inputState.gamepadUnavailableWarned.has(i)) {
+                        this.inputState.gamepadUnavailableWarned.add(i);
+                        console.warn("Your browser does not know the mapping for the %s gamepad (index %d) and we can't use it.", gamepad.id, i);
```

Comment:
It would be even nicer to use `this.notifications.show()`, which shows a text notification to the player without them needing to check the JS console:

```suggestion
                        this.notifications.show("Unsupported gamepad: " + gamepad.id);
```

Reply:
(Reduced the amount of text in this error message only because the font for the notifications is pretty huge) 😄

Reply:
Done.